lightning-invoice/src/utils.rs

```diff
@@ -218,6 +215,81 @@ where
 	}
 }
 
+/// Utility to filter channels for an invoice, and return the corresponding route hints to include
+/// in the invoice.
+///
+/// The filtering ensures that the RouteHints returned only include the highest inbound capacity
+/// channel for the pubkey returned by the map_channels_to_pubkey. Channels with a lower inbound
+/// capacity than the invoice amount aren't included in the hints.
+///
+/// `channels`: The channels to filter.
+/// `min_inbound_capacity_required`: Defines the lowest inbound capacity a channel must have to not
+/// be filtered out.
+/// `empty_result_if_private_channels_exists`: If set to `true`, this entire function will return
+/// zero route hints if any of the inputted channels is a private channel.
+/// `channel_pubkey_selector`: Function that is called for every channel evaluated in the filtering
+/// that the required fields and has a high enough inbound_capacity. The route hint for the
+/// highest inbound capacity channel will be returned for each unique `PublicKey` returned by
+/// `channel_pubkey_selector`. The `ChannelDetails` parameter in `channel_pubkey_selector`, is the
+/// current channel evaluated for filtering.
+/// `route_hint_from_channel`: Mapping function that defines how a channel should be mapped into a
+/// `RouteHint`. The `ChannelDetails` parameter in `channel_pubkey_selector`, is the current channel evaluated
+/// for filtering. The `u64` parameter is the channel's `short_channel_id`.
+fn filter_channels<F1, F2>(
+	channels: Vec<ChannelDetails>,
+	min_inbound_capacity_required: Option<u64>,
+	empty_result_if_private_channels_exists: bool,
```

Comment:
Hmm, I'm not sure if this captures what we want exactly, if our channels are public and we're creating a phantom route hint, instead of including all the channels, we can instead just create route hints that are one hop instead of two. id instead of creating the route hint in a callback, we can instead have the `create_phantom_invoice` callsite push a second hop onto all the route hints we get (or, if we get none, create a single route hint that is just one hop long). Then we could remove this option, I believe.

Reply:
Thanks! If my new understanding explained in the logic below is correct, we can indeed remove it.